backend/backend.go

```diff
@@ -540,6 +544,97 @@ func (backend *Backend) Accounts() AccountsList {
 	return backend.accounts
 }
 
+// AccountsByKeystore returns a map of the current accounts of the backend, grouped
+// by keystore.
+func (backend *Backend) AccountsByKeystore() (KeystoresAccountsListMap, error) {
+	accountsByKeystore := KeystoresAccountsListMap{}
+	defer backend.accountsAndKeystoreLock.RLock()()
+	for _, account := range backend.accounts {
+		persistedAccount := account.Config().Config
+		rootFingerprint, err := persistedAccount.SigningConfigurations.RootFingerprint()
+		if err != nil {
+			return nil, err
+		}
+		keystore, err := backend.Config().AccountsConfig().LookupKeystore(rootFingerprint)
+		if err != nil {
+			return nil, err
+		}
+		accountsByKeystore[keystore] = append(accountsByKeystore[keystore], account)
+	}
+	return accountsByKeystore, nil
+}
+
+// KeystoreTotalAmount represents the total balance amount of the accounts belongings to a keystore.
+type KeystoreTotalAmount = struct {
+	// FiatUnit is the fiat unit of the total amount
+	FiatUnit string `json:"fiatUnit"`
+	// Total formatted for frontend visualization
+	Total string `json:"total"`
+}
+
+// AccountsTotalBalanceByKeystore returns a map of accounts' total balances across coins, grouped by keystore.
+func (backend *Backend) AccountsTotalBalanceByKeystore() (map[string]KeystoreTotalAmount, error) {
```

Comment:
maybe also move to accounts.go. unit test would be nice here too :P

Reply:
Partially addressed in #2609 (unit test will come in a separate PR)